homeassistant/components/nextcloud/sensor.py

```diff
@@ -0,0 +1,111 @@
+"""Summary data from Nextcoud."""
+from datetime import timedelta
+import logging
+
+from nextcloudmonitor import NextcloudMonitor
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_URL,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+# Assign domain
+DOMAIN = "nextcloud"
+# Set Default scan interval in seconds
+SCAN_INTERVAL = timedelta(seconds=60)
+
+# Validate user configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
It's better to move the config to the component module in the integration. That makes it more future proof.

The platforms can be set up via `discovery.load_platform`.

Reply:
@MartinHjelmare Is there an example of this somewhere that I can use as a starting point?